### ISSUE TITLE
Add support for Servlets

### ISSUE THREAD

**Issue by roti** [gcv/appengine-magic#53]
Hi,

This is a feature request to add support for standard java web applications. I am in the situation where I need to start the appengine environment with Java servlets and web.xml style configuration. As I have seen currently appengine-magic currently only works with ring handlers.

Regards,
Razvan

**Comment by gcv** [comment]
appengine-magic should work fine in conjunction with Java servlets. When you run `lein appengine-new`, the task generates a `war/WEB-INF/web.xml` file pre-filled with everything necessary for the Ring handler, as a servlet. You can add other servlet handlers to the configuration. You'll have to figure out how to link building your Java and your Clojure code, but perhaps Leiningen can help there.

Let me know if you already tried this approach and it didn't work.

**Comment by roti** [comment]
Well, I'm not using the ring handler. I have a java servlet to do the job. I want to use appengine-magic for the other services (like datastore). Currently appengine-magic.core/serve requires an "application" as parameter.

Razvan

**Comment by gcv** [comment]
Right, appengine-magic does need to be started in order to provide Google's services. Why don't you just make a placeholder application which does not serve up any HTTP requests? You don't even need to have it mapped in web.xml.

**Comment by roti** [comment]
Not sure exactly what you mean by the placeholder app. Anyway, I took a more daring path: I tried to start an embedded dev_appserver (which is started by dev_appserver.sh, from the appengine sdk), but ran into some problems. The security manager does not allow reading of web.xml files. I posted a question on the appengine forum and am currently waiting for some help.

Maybe it would be a good idea to integrate this into appengine-magic: to start the real dev_appserver with the start method, instead of a blank jetty instance. With this I think you won't need to do any of those "/_ah/*" mappings. What do you think? I'm willing to share the code once I get the thing right.

Razvan

**Comment by gcv** [comment]
I tried to figure out what dev_appserver does and start it sometime in September 2010, ran into massive problems, and did what you see in appengine-magic today.

Rereading the thread, I finally realized that you're having trouble with using interactive mode. I see. The easiest solution is for you to modify `appengine-magic.jetty` to respect extra servlets mapped in `web.xml`. You'll have to figure out the right way of doing this from the Jetty API, and there are several reasonable approaches.
